### DISCUSSION TITLE
How to prevent bonds from passing through each other?

### DISCUSSION THREAD

**Discussion by nmendiboure** [glotzerlab/hoomd-blue#1904] [Support] [answered]
![demo](https://github.com/user-attachments/assets/23de53bc-5db8-430b-988d-ec1847e333e2)

Hi,

I'm running a coarse-grained simulation where I model polymers with harmonic bonds and harmonic angles. I'm also using a shifted Lennard-Jones potential for non-bonded pairwise interactions and a Langevin thermostat to control the temperature.

However, I sometimes observe that covalent bonds pass through other bonds, which is non-physical. I've tried adjusting the bond stiffness and the parameters of the Lennard-Jones potential, but the issue persists. Is there something I can do to prevent this behavior? Should I consider changing the type of bonds or the potentials used in the system?

Thanks in advance

**Comment by joaander** [comment, marked as answer]
[FENE](https://hoomd-blue.readthedocs.io/en/v4.8.2/module-md-bond.html#hoomd.md.bond.FENEWCA) bonds have a maximum length at which the energy goes to infinity. Harmonic bonds can stretch to any length. You can read about this model in just about any coarse-grained polymer model in the literature.

HOOMD-blue implements FENE+WCA because some polymer models use LJ for non-bonded pairwise interactions and WCA for bonded. If you use LJ for all (i.e. you are disabling the default "bond" exclusion in the neighbor list), then you should set the WCA part of FENEWCA to 0.

**Reply by nmendiboure** [reply]
Thanks ! I replaced harmonic bonds by FENEWCA (with epsilon and sigma to 0 and empty exclusion list) and its seems to work properly.
